docs/source/managed-federation/uplink.mdx

```diff
@@ -0,0 +1,90 @@
+
+```mermaid
+graph LR;
+  subgraph "Your infrastructure"
+  serviceA[Products subgraph];
+  serviceB[Reviews subgraph];
+  gateway([Gateway]);
+  end
+  subgraph "Apollo cloud"
+  registry{{Apollo Schema Registry}};
+  uplink{{Apollo Uplink}}
+  end
+  serviceA & serviceB -->|Pushes schema| registry;
+  registry -->|Updates config| uplink;
+  gateway -->|Polls for config changes| uplink;
+  class registry secondary;
+  class uplink secondary;
+```
+
+To maximize uptime, Uplink is hosted simultaneously at _two_ endpoints, one in GCP and one in AWS:
+
+* GCP: `https://uplink.api.apollographql.com/`
+* AWS: `https://aws.uplink.api.apollographql.com/`
+
+By default in versions of `@apollo/gateway` vTODO and later, your gateway polls the GCP URL at a regular interval. Whenever a poll request fails, the gateway retries the request with the AWS URL. It continues swapping between the two URLs until the request succeeds, or until reaching the maximum number of retries.
+
+> Verisons of `@apollo/gateway` prior to vTODO don't support multiple Uplink endpoints and only use the GCP URL by default.
+
+## Configuring polling behavior
+
+You can configure the list of Uplink URLs your gateway uses, along with the number of retries it performs for a failed poll request.
+
+### Uplink URLs
+
+You can provide a custom list of URLs for the gateway to use when polling Uplink. For example, you can prioritize the AWS URL or omit one of the URLs entirely.
+
+You can provide this list either in the `ApolloGateway` constructor or as an environment variable.
+
+#### `ApolloGateway` constructor
+
+Provide a custom list of Uplink URLs to the `ApolloGateway` constructor like so:
+
+```js
+const { ApolloGateway } = require('@apollo/gateway');
+
+// ...
+
+const gateway = new ApolloGateway({
+  uplinkEndpoints: [
+    'https://aws.uplink.api.apollographql.com/',
+    'https://uplink.api.apollographql.com/'
+  ]
+});
+```
+
+This example swaps the priority of the two Uplink URLs, which means the gateway polls the AWS URL first.
+
+Note that if you _also_ provide a list of endpoints via [environment variable](#environment-variable), the environment variable takes precedence.
+
+#### Environment variable
+
+You can provide a comma-separated list of Uplink URLs as the value of the `APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT` environment variable in your gateway's environment.
+
+For example, this configuration polls the AWS URL exclusively:
+
+```bash
+APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT=https://aws.uplink.api.apollographql.com/
+```
+
+### Retry limit
+
+You can configure how many times your gateway retries a failed poll request like so:
+
+```js
+const { ApolloGateway } = require('@apollo/gateway');
+
+// ...
+
+const gateway = new ApolloGateway({
+  uplinkMaxRetries: 2
+});
+```
+
+By default, the gateway retries a number of times equal to three times the number of [Uplink URLs](#uplink-urls).
```

Comment:
I think we should make it clear that this is only an _internal_ retry, in that we will keep polling the endpoints in the list this number of times for _one fetch_ before emitting an error. We will always poll indefinitely when there are errors for all retries.

Reply:
In other words, if you hit `maxRetries` while round-robin-ing through the endpoints, the behavior is unchanged from today. You still will continue to use the last fetched supergraph and try again at the next poll interval.